### PR TITLE
static_init! cleanup

### PR DESCRIPTION
**Pull request by bradjc**
Making progress towards using the updated `static_init!` API.

### CODE REVIEW

File: boards/imix/src/main.rs [hunk 1]
```diff
@@ -320,38 +303,31 @@ pub unsafe fn reset_handler() {
...
-        [&'static sam4l::gpio::GPIOPin; 8],
+        [&'static sam4l::gpio::GPIOPin; 7],
         [&sam4l::gpio::PC[31], // P2
          &sam4l::gpio::PC[30], // P3
          &sam4l::gpio::PC[29], // P4
          &sam4l::gpio::PC[28], // P5
          &sam4l::gpio::PC[27], // P6
          &sam4l::gpio::PC[26], // P7
          &sam4l::gpio::PA[20], // P8
-         &sam4l::gpio::PA[20], // Dummy Pin (regular GPIO)
```
Comment (ppannuto) [inline boards/imix/src/main.rs]: What was this? And should deleting it be part of updating the use of the `static_init` macro?

Comment (bradjc) [inline boards/imix/src/main.rs]: I don't know. It was brought up in the imix PR, but didn't get addressed there. The commit message for that commit mentions general updates to imix.

**Comment by brghena** [comment]
Woah, is it valid to use `static_init!` inside capsules? I didn't know we ever did that.

**Comment by bradjc** [comment]
We don't. Those are in comments as example usages.

File: boards/imix/src/main.rs (hunk 1 above)
Comment (alevy) [inline boards/imix/src/main.rs]: I don't know why this is still here. I _believe_ it was a hack for the rf233 user library when it was a thing. I also _believe_ it's not useful anymore. Let's remove it silently and see if anyone yells.

**Pull request merged**